src/main/java/ca/nrc/cadc/beacon/web/config/VOSpaceServiceConfig.java

```diff
@@ -0,0 +1,124 @@
+ *                                       par écrit.
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+package ca.nrc.cadc.beacon.web.config;
+
+import java.net.URI;
+import org.springframework.util.Assert;
+
+public class VOSpaceServiceConfig {
+    private String name;
+    private String properName;
+    private String serviceType;
+    private URI resourceID;
+    private URI nodeResourceID;
+
+    // Default provided, can be overridden
+    public String homeDir;
+
+    public VOSpaceServiceConfig(String name, String properName,
+                                String serviceType, URI resourceID,
+                                URI nodeResourceID) {
+
+        Assert.hasLength(name, "VOSpace service name required");
+        Assert.hasLength(properName, "VOSpace service proper name (ie 'arc' or 'vos') required");
+        Assert.hasLength(serviceType, "VOSpace service type (ie 'cavern' or 'vault') required");
+        Assert.notNull(resourceID, "service resource ID required");
+        Assert.notNull(nodeResourceID, "node resource ID required");
```

Comment:
These can be empty URIs, too.

Reply:
Changed how this section is done, added a unit test for the class to make sure checks & throws are done properly.